gems/smithy-client/lib/smithy-client/net_http/connection_pool.rb

```diff
@@ -0,0 +1,279 @@
+# frozen_string_literal: true
+
+require 'cgi'
+require 'net/http'
+require 'net/https'
+require 'delegate'
+require 'logger'
+
+require_relative 'patches'
+
+Smithy::Client::NetHTTP::Patches.apply!
+
+module Smithy
+  module Client
+    module NetHTTP
+      # @api private
+      class ConnectionPool
+        @pools_mutex = Mutex.new
+        @pools = {}
+
+        OPTIONS = {
```

Comment:
I noticed that the `Hearth::HTTP::Client` had some values set. Does any of them apply here? Linked reference here: https://github.com/smithy-lang/smithy-ruby/blob/main/hearth/lib/hearth/http/client.rb#L13

Reply:
I think using the net http defaults is probably preferred. We mostly did that before. I think those values are also the same defaults anyway.